backend/rate_limiter.py

```
@@ -0,0 +1,313 @@
+        if last_error:
+            raise last_error
+
+    async def _process_queue(self) -> None:
+        """Process queued requests."""
+        while True:
+            try:
+                await asyncio.sleep(0.1)
+                # Queue processing logic for future use
+            except asyncio.CancelledError:
+                break
+            except Exception as e:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)